cartography/intel/aws/iam.py

```diff
@@ -309,11 +309,15 @@ def load_roles(
     neo4j_session: neo4j.Session, roles: List[Dict], current_aws_account_id: str, aws_update_tag: int,
 ) -> None:
     ingest_role = """
-    MERGE (rnode:AWSRole{arn: $Arn})
-    ON CREATE SET rnode:AWSPrincipal, rnode.roleid = $RoleId, rnode.firstseen = timestamp(),
-    rnode.createdate = $CreateDate
-    ON MATCH SET rnode.name = $RoleName, rnode.path = $Path
-    SET rnode.lastupdated = $aws_update_tag
+    MERGE (rnode:AWSPrincipal{arn: $Arn})
```

Comment:
Moving things from the `ON CREATE SET` and the `ON MATCH SET` is good.

However I don't think we should merge by the principal first. From #1133, the only situations in cartography where we create these "bare/husk/whatever a good name for them is" principal nodes are

1. When we [parse AssumeRoleDocuments](https://github.com/lyft/cartography/blob/d94ef53cbf5b23ecf132d472a8952a9cad1b89ce/cartography/intel/aws/iam.py#L325) -- in this case, we should definitely set the `AWSRole` label there, because assumerole only happens for roles and not users or groups.
2. When we [handle redshift IAM role attachments](https://github.com/lyft/cartography/blob/d94ef53cbf5b23ecf132d472a8952a9cad1b89ce/cartography/intel/aws/redshift.py#L117) -- in this case, we should also definitely set the `AWSRole` label there because the field in the API data is called ["IamRoleArn"](https://github.com/lyft/cartography/blob/d94ef53cbf5b23ecf132d472a8952a9cad1b89ce/cartography/intel/aws/redshift.py#L126), so it wouldn't make sense to keep that as a bare principal.

Reply:
We probably should be setting those as Roles, but It's also account roots, users, and federated identities that can assume roles.

- https://aws.amazon.com/blogs/security/how-to-use-trust-policies-with-iam-roles/
- https://stackoverflow.com/questions/66414774/can-iam-user-assume-a-role-if-the-user-is-not-described-in-the-trust-policy-of

Reply:
We talked about this:

- We aren't going to parse ARNs yet to determine the specific principal type. This will come in a future PR.
- When syncing trust relationships, we don't know the type of the trusted principal unless we parse the ARN. Since we aren't doing that yet, we will continue to merge with the generic label -- that is, we will continue to do `merge :AWSPrincipal` instead of `merge :<specific label>`.

Therefore when we reach sync_roles, we will do `MERGE :AWSPrincipal` to solve #1133.